src/sample/courseRegController.java

```diff
@@ -48,67 +58,226 @@ public void ContactUs(MouseEvent event) throws IOException {
+    if(courseList.getSelectionModel().getSelectedItem() != null){
+      //Store Courses Registered into txt file
+      try (FileWriter myWriter = new FileWriter("student1registeredCourse.txt",true)){
+        myWriter.write(String.valueOf(courseList.getSelectionModel().getSelectedItem()), 0,
+        String.valueOf(courseList.getSelectionModel().getSelectedItem()).length());
+        myWriter.write("\n");
+        //System.out.println("Successfully wrote to the file.");
+      } catch (IOException e) {
+        System.out.println("An error occurred.");
+        e.printStackTrace();
+      }
+    
+      //Clear the Registered Course ListView
+    registeredCourse.getItems().clear();
+
+    //Read and Display Courses Registered from txt file
+    displaycourseregistered();
+
+    //Filter out courses that have been registered by the student.
+    filtercourselist();
+
+    //Clear the Course List ListView
+    courseList.getItems().clear();
+
+    //Read and display list of all courses that can be registered from txt file
+    displaycourselist();
+    
+    //Clear the searchCourse Textfield after a course is registered.
+    searchCourse.clear();
+    } 
+  }
+
+  public void removecourse()  {
+    //Remove Courses Registered from txt file
+    if(registeredCourse.getSelectionModel().getSelectedItem() != null){      
+      try{
+        File file = new File("student1registeredCourse.txt");
+        File temp = new File("TempFile.txt");
+        //File temp = File.createTempFile("temporarystudent", ".txt", file.getParentFile());
+        BufferedReader reader = new BufferedReader(new FileReader(file));
+        BufferedWriter writer = new BufferedWriter(new FileWriter(temp));
+
+        String lineToRemove = registeredCourse.getSelectionModel().getSelectedItem();
+        String currentLine;
+
+        while((currentLine = reader.readLine()) != null) {
+          // trim newline when comparing with lineToRemove
+          String trimmedLine = currentLine.trim();
+          if(trimmedLine.equals(lineToRemove)) continue;
+          writer.write(currentLine + System.getProperty("line.separator"));
+        }
+
+        //Close the reader and writer (preferably in the finally block).
+        reader.close();
+        writer.close();
+        //Delete the file.
+        file.delete();
+        //Rename the temp file.
+        temp.renameTo(file);
+        
+      } catch (IOException e) {
+        System.out.println("An error occurred.");
+        e.printStackTrace();
+      }
+      //Clear the Registered Course ListView
+      registeredCourse.getItems().clear();
+      //Read and Display Courses Registered from txt file
+      displaycourseregistered();
+
+      //Filter out courses that have been registered by the student.
+      filtercourselist();
+
+      //Clear the Course List ListView
+      courseList.getItems().clear();
+
+      //Read and display list of all courses that can be registered from txt file
+      displaycourselist();
```

Comment:
```
registeredCourse.getItems().clear();
      //Read and Display Courses Registered from txt file
      displaycourseregistered();

      //Filter out courses that have been registered by the student.
      filtercourselist();

      //Clear the Course List ListView
      courseList.getItems().clear();

      //Read and display list of all courses that can be registered from txt file
      displaycourselist();
```
Almost the same as the one in `addcourse()`, can make a function to improve reusability